x/superfluid/keeper/stake.go

```diff
@@ -503,17 +504,34 @@ func (k Keeper) forceUndelegateAndBurnOsmoTokens(ctx sdk.Context,
 	if err != nil {
 		return err
 	}
-	// TODO: Better understand and decide between ValidateUnbondAmount and SharesFromTokens
-	// briefly looked into it, did not understand what's correct.
-	// TODO: ensure that intermediate account has at least osmoAmount staked.
+
 	shares, err := k.sk.ValidateUnbondAmount(
 		ctx, intermediaryAcc.GetAccAddress(), valAddr, osmoAmount,
 	)
 	if err == stakingtypes.ErrNoDelegation {
 		return nil
-	} else if err != nil {
-		return err
 	}
+
+	if err != nil {
+		// if ValidateUnbondAmount has failed it indicates that the amount we're trying to unbond is
+		// greater then what validator has this can be due to different factors (e.g jail)
```

Comment:
```suggestion
		// greater then what validator has delegated to it. This can be due to different factors (e.g jail)
```

This actually doesn't quite check out to me as an explanation. If a validator is downtime jailed on Osmosis, there is no slash. So why would the unbond amount tracked be greater than expected post jail?